pages/categorization.vue

```diff
@@ -117,49 +110,143 @@
+
+					// A. Handle changes for when at least one category is painted
+					case this.possibleStates.STATE_ATLEASTONE_CATEGORY_PAINTED:
+						this.changeState_AtLeastOneCategoryPainted();
+						break;
+
+					// B. Handle changes for no categories are painted
+					default:
+						this.changeState_NoCategoriesPainted();
+						break;
+				}
+			},
+
+			changeState_AtLeastOneCategoryPainted() {
+
+				// 1. Paint the table with previously painted rows documented in the store
+				this.columnLinkingTable.needsRefresh = true;
+
+				// Enable access to the annotation page
+				this.nextPageAccess(true);
+			},
+
+			changeState_NoCategoriesPainted() {
+
+				// 1. Paint the table with previously painted rows documented in the store
+				this.columnLinkingTable.needsRefresh = true;
+				// this.$refs.table.$forceUpdate();
+
+				// Disable access to the annotation page
+				this.nextPageAccess(false);
+			},
+
+			changeToNewState() {
+
+				// 1. Determine page state from data contents
+				let newState = this.determineState();
+
+				// 2. Change the page to the determined state
+				this.changeState(newState);
+			},
+
+			countLinkedColumns() {
+
+				// Count the number of columns that have had categories linked to them
+				let links = 0;
+				for ( let column in this.columnToCategoryMap ) {
+					if ( null != this.columnToCategoryMap[column] )
+						links += 1;
+				}
+
+				return links;
+			},
+
+			determineState() {
+
+				// 1. Reset the state to default
+				let newState = this.possibleStates.STATE_NOCATEGORIES_PAINTED;
+
+				// 2. Count the number of painted rows in the table
+				let paintedRowsCount = this.countLinkedColumns();
+
+				// 3. Add appropriate state flags based on data contents
+				if ( paintedRowsCount > 0 )
+					newState |= this.possibleStates.STATE_ATLEASTONE_CATEGORY_PAINTED;
+
+				return newState;
+			},			
+
+			nextPageAccess(p_enable) {
 				
-				// Local reference to the page names in the store
-				pageData: this.$store.getters.pageData,
+				// 1. Enable/disable access to the annotation page on the nav bar
 
-				// Whether or not page has enabled access to the annotation page
-				readyForNextStepFlag: false,
+				// A. Enable the nav item
+				this.$store.dispatch("enablePageNavigation", { 
+					pageName: "annotation",
+					enable: p_enable
+				});
 
-				// Data for the coloring listgroup
-				recommendedCategories: [
+				// B. Change the next step button's color
+				this.nextPageButtonColor = ( p_enable ) ? "success" : "secondary";
+			},			
 
-					"Subject ID",
-					"Age",
-					"Sex",
-					"Diagnosis",
-					"Assessment Tool"
-				],
+			resetTableRefresh() {
 
-				tableID: "category-painting-table"
-			}
-		},
+				this.needsRefresh = false;
+			},
 
-		computed: {	
+			setSelectedCategory(p_clickData) {
 
-			tableDataFromTsvAndOrJson() {
+				// Save the name of the selected category
+				this.selectedCategory = p_clickData.category;
+			},
+			
+			setupColumnToCategoryTable() {
```

Comment:
@surchs Since I'm committing small changes before merging this PR as discussed, please write up your suggestions for this function in a new issue and assign it to me.

Reply:
Hey @jarmoza, I already did. That's what I meant by making the suggestions as a commit. You would have to pull the PR branch to see these changes locally!

Reply:
Okay. I will check for this change in a bit.